Math/NumberTheory/ArithmeticFunctions/Standard.hs

```diff
@@ -105,22 +107,22 @@ tau = runFunction tauA
 tauA :: Num a => ArithmeticFunction n a
 tauA = multiplicative $ const (fromIntegral . succ)
 
-sigma :: (UniqueFactorisation n, Integral n) => Word -> n -> n
+sigma :: (UniqueFactorisation n, Integral n, Integral m) => Word -> n -> m
```

Comment:
I'd rather stick to the previous type signature here and below. Truly, any function `f :: Integral n :: n -> n` can be generalised to `f' :: (Integral n, Integral m) :: n -> m`, but usually there are two reasons to avoid it.

1. Too polymorphic types make type inference feeble. E. g., 

    ```haskell
    sigmaOfSigma :: Integer -> Integer 
    sigmaOfSigma = sigma 1 . sigma 1
    ```

    perfectly compiles for `sigma :: Integral n :: Word -> n -> n`, but fails for `sigma :: (Integral n, Integral m) :: Word -> n -> m`, because GHC does not know which type is emitted by inner `sigma` and consumed by outer one: any `Integral` fits the bill.

2. It is unclear when type conversion actually takes place. There are at least three possibilities: `sigma :: (Integral n, Integral m) :: Word -> n -> m` may

    1. cast its argument `n` to type `m` at the very beginning and then proceed as usual: `sigma a = oldSigma a . fromIntegral`.
    
    2. proceed as usual and cast its result `n` to type `m` at the very end: `sigma a = fromIntegral . oldSigma a`.

    3. cast `n` to `m` during intermediate computations. E. g., in your patch conversion takes place inside `sigmaHelper`.

    If `n ~ Integer` and `m ~ Int`, then each of options above may lead to different results due to overflow will happen at different stages. If `sigma` is less polymorphic (`sigma :: Integral n :: Word -> n -> n`), it becomes clear for an end user where a cast takes place, because he is obliged to insert it manually.

Reply:
Sticking to the previous type signature for `sigma` is possible, but for `ramanujanTau` it's unclear to me how this would work - it would cause `ramanujanTau (2 :: Natural)` to create arithmetic underflow, since the output should be negative.  In this way, I'd argue the generalisation of `ramanujanTau` is necessary, unless the underflow behaviour is preferred.

Reply:
IMHO the most reasonable output type of `ramanujan` is simply `Integer`. Its value grows too rapidly to be confined within bounds of `Int`: `ramanujan 2647 > 2 ^ 63`. And we cannot detect overflow by rapid flip of sign, because `ramanujan` changes sign frequently. So we need a signed arbitrary-precision `Integral` - and basically only `Integer` satisfies these conditions. 

The input of `ramanujan` can be any `Integral`, but I'd rather stick to arbitrary-precision ones. The internals of the algorithm are not opaque for an end user and he cannot predict, whether an overflow may (or is likely to) happen, especially when eleventh powers get involved. That said, the input type should be either `Integer` or `Natural`, but we do not have any standard Haskell class to cover them. For the sake of simplicity, it is not unreasonable to define simply `ramanujan :: Integer -> Integer`.

Reply:
Sounds good, I've amended.